parachain/primitives/router/src/outbound/mod.rs

```diff
@@ -225,11 +264,34 @@ impl<'a, Call> XcmConverter<'a, Call> {
 			UnpaidExecution { check_origin: None, weight_limit: Unlimited } => None,
 			_ => return Err(TargetFeeExpected),
 		};
+		log::trace!(target: "xcm::ethereum_blob_exporter", "execution_fee is {execution_fee:?}.");
 		Ok(execution_fee)
 	}
 
+	fn is_transact(&self) -> bool {
+		for instruction in self.iter.clone() {
+			// TODO not certain if this is OK to differentiate between TransferToken and Transact
+			// messages
+			if let Transact { .. } = instruction {
```

Comment:
@yrong I need to distinguish between a `TransferToken` and `Transact` message in the router. Here, I just look if there is a Transact message in the XCM messages. I am not sure this is correct, please let me know.

Reply:
Should be fine as a workaround since there is not much `AgentExecuteCommand` to distinguish, and it's unlikely to include a `Transact` into the `Transfer*` Command.

I'll also check if there is some xcm primitive for user customized data(like the type of Command here).